tools/smoke_test.py

```diff
@@ -0,0 +1,174 @@
+            f"- minimum requirement: {MIN_TORCH_VERSION}"
+        )
+
+
+def check_cuda():
+    import torch
+
+    if not torch.cuda.is_available():
+        sys.stderr.write("CUDA not available for torch\n")
+        return
+
+    torch_cuda_ver = Version(torch.version.cuda)
+    if torch_cuda_ver < MIN_CUDA_VERSION:
+        raise RuntimeError(
+            f"CUDA version not supported: {torch_cuda_ver} "
+            f"- minimum requirement: {MIN_CUDA_VERSION}"
+        )
+
+    # check if torch cuda version matches system cuda version
+    cuda_home = os.getenv("CUDA_HOME")
+    if not cuda_home:
+        raise RuntimeError("$CUDA_HOME not found")
+
+    nvcc = os.path.join(cuda_home, "bin", "nvcc")
+    nvcc_version_str = (
+        subprocess.check_output([nvcc, "--version"]).strip().decode("utf-8")
+    )
+    cuda_version = re.search(r"release (\d+[.]\d+)", nvcc_version_str)
+    if cuda_version is None:
+        raise RuntimeError("CUDA version not found in `nvcc --version` output")
+
+    cuda_ver = Version(cuda_version.group(1))
+    if cuda_ver != torch_cuda_ver:
+        raise RuntimeError(
+            f"CUDA version mismatch, torch version: {torch_cuda_ver}, env version: {cuda_ver}"
+        )
+
+
+def check_dynamo(backend, device, err_msg):
+    import torch
+
+    if device == "cuda" and not torch.cuda.is_available():
+        sys.stderr.write(
+            f"CUDA not available -- skipping CUDA check on {backend} backend\n"
+        )
+        return
+    try:
+        import torchdynamo
+
+        torchdynamo.reset()
+
+        @torchdynamo.optimize(backend, nopython=True)
+        def fn(x):
+            return x + x
+
+        x = torch.randn(10, 10).to(device)
+        x.requires_grad = True
+        y = fn(x)
+        torch.testing.assert_close(y, x + x)
+    except Exception:
+        sys.stderr.write(traceback.format_exc() + "\n" + err_msg + "\n\n")
+        sys.exit(1)
+
+
+def check_eager_cpu():
+    check_dynamo(
+        "eager",
+        "cpu",
+        "CPU eager sanity check failed",
+    )
+
+
+def check_eager_cuda():
+    check_dynamo(
+        "eager",
+        "cuda",
+        "CUDA eager sanity check failed",
+    )
+
+
+def check_aot_eager_cpu():
+    check_dynamo(
+        "aot_eager",
+        "cpu",
+        "CPU aot_eager sanity check failed",
+    )
+
+
+def check_aot_eager_cuda():
+    check_dynamo(
+        "aot_eager",
+        "cuda",
+        "CUDA aot_eager sanity check failed",
+    )
+
+
+def check_inductor_cpu():
+    check_dynamo(
+        "inductor",
+        "cpu",
+        "CPU inductor sanity check failed",
+    )
+
+
+def check_inductor_cuda():
+    check_dynamo(
+        "inductor",
+        "cuda",
+        "CUDA inductor sanity check failed\n"
+        + "NOTE: Please check that you installed the correct hash/version of `triton`",
+    )
+
+
+def main():
+    check_python()
+    check_pip_deps()
+    check_torch()
+    check_cuda()
```

Comment:
For all these checks above, it's better we can print out the version details even if they pass all checks. Users can leverage this script to collect env info when they reporting bugs, and help us to narrow down where is the bug from.